app/src/main/java/a3/com/convo/adapters/FriendAdapter.java

```diff
@@ -53,61 +66,35 @@ public FriendAdapter(ArrayList<String> friends) {
     @Override
     public void onBindViewHolder(@NonNull final FriendAdapter.ViewHolder holder, int position) {
         holder.itemView.setBackgroundColor(selectedPos == position ? Color.rgb(229,229,229) : Color.TRANSPARENT);
-        // Get friends user ID's
-        final String friendId = myFriends.get(position);
-        final String profilePicPath = "/" + friendId + "/picture";
-
-        //Get friends profile picture
-//        GraphRequest picRequest = GraphRequest.newGraphPathRequest(AccessToken.getCurrentAccessToken(), "/" + friendId + "/picture", new GraphRequest.Callback() {
-//            @Override
-//            public void onCompleted(GraphResponse response) {
-//                try {
-//
-//                } catch (JSONException e) {
-//
-//                }
-//            }
-//        });
-
-        GraphRequest picRequest = GraphRequest.newMeRequest(
-                AccessToken.getCurrentAccessToken(),
-                new GraphRequest.GraphJSONObjectCallback() {
-                    @Override
-                    public void onCompleted(
-                            JSONObject json_object,
-                            GraphResponse response) {
-                        try {
-                            JSONObject picInfo = json_object.getJSONObject("picture").getJSONObject("data");
-                            String picURL = picInfo.optString("url");
-                            GlideApp.with(context)
-                                    .load(picURL)
-                                    .circleCrop()
-                                    .into(holder.ivFriend);
-                        } catch(Exception e){
-                            e.printStackTrace();
-                            Toast.makeText(context, "Pic URL not retrieved", Toast.LENGTH_SHORT).show();
-                        }
+        // Get friend
+        final String friend = myFriends.get(position);
```

Comment:
Can position be invalid or null?

Reply:
I actually don't understand this one. When it comes to clicking on the RV, I checked if the position is invalid or not, ut when it comes to the positions in the array displaying on the RV, I'm not sure why any of the positions would be null, as the "friends" are created upon login. Would it be good to add a check in any case? PS: Our "myFriends" array are friends that you have on Facebook that are registered to Convo.

Reply:
We should always think about edge cases. What happens if server data is wiped out? We should always handle that on client. What happens if the user does not have any friends

Reply:
Lets add a check if position is null or if the array size if more than the position